File: 0110-balanced-binary-tree/0110-balanced-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isBalanced(TreeNode root) {
        if(root == null) return true;

        return check(root) != -1;
    }
    private int check(TreeNode node){
        if(node == null) return 0;

        int lH = check(node.left);
        int rH = check(node.right);

        if(lH == -1 || rH == -1) return -1;
        if(Math.abs(lH - rH) > 1) return -1;

        return Math.max(lH, rH) + 1;
        
    }
}

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isBalanced(root *TreeNode) bool {
    if root == nil { return true }

    return rec(root) != -1
}

func rec(node *TreeNode) int {
    if node == nil { return 0 }

    l := rec(node.Left)
    r := rec(node.Right)

    if l == -1 || r == -1 { return -1 }
    if Abs(l - r) > 1 { return -1 }

    return Max(l, r) + 1
}

func Abs(num int ) int {
    if num < 0 { return num * (-1)}
    return num
}

func Max(a, b int) int {
    if a > b { return a }
    return b
}